fv_be/app/admin/base_admin.py

```diff
@@ -70,16 +71,30 @@ def item_id(self, instance):
     item_id.short_description = _("Id")
 
     def admin_link(self, instance):
-        url = reverse(
-            f"admin:{instance._meta.app_label}_{instance._meta.model_name}_change",
-            args=(instance.id,),
-        )
-        # todo: i18n for 'Edit' not working here for some reason
-        return format_html('<a href="{}">{}: {}</a>', url, "Edit", str(instance))
+        # todo: Add reverse links for all models
+        try:
+            url = reverse(
+                f"admin:{instance._meta.app_label}_{instance._meta.model_name}_change",
+                args=(instance.id,),
+            )
+            # todo: i18n for 'Edit' not working here for some reason
+            return format_html('<a href="{}">{}: {}</a>', url, "Edit", str(instance))
+        except NoReverseMatch as e:
+            return None
 
     def save_model(self, request, obj, form, change):
         if not change:
             obj.created_by = request.user
 
         obj.last_modified_by = request.user
         super().save_model(request, obj, form, change)
+
+
+# todo: better name for following class
+class AdminHideUtility(BaseAdmin):
+    """
+    This utility class lets the admin classes to be registered on the top level of admin interface, to get the reverse
+    links working for inline admin classes and also prevents the classes that inherit this to be shown at the top level.
+    """
+    def get_model_perms(self, request):
+        return {}
```

Comment:
Can you please verify that this doesn't remove permission checking on the actual edit forms? E.g., staff should not have access to edit everything that superadmins can?

Reply:
Yes, I believe it does not remove permission checking on the edit forms.
I tested above by creating one more user (lets call it nonAdmin), gave it only permission to add a site and a dictionaryEntry and not other models. The nonAdmin then was only able to do the permitted stuff and was not able to CRUD on any other models which have also inherited from this class.